Mathlib/Data/Nat/Prime/EulerFermat.lean

```diff
@@ -0,0 +1,36 @@
+/-
+Copyright (c) 2024 Ralf Stephan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ralf Stephan, Thomas Browning
+-/
+import Mathlib.Data.Nat.Prime.Basic
+import Mathlib.Data.Nat.Factorization.Basic
+
+/-!
+
+# Primality of `a ^ n + 1`
+
+Euler proved around 1738 that some Fermat numbers are prime, but not `F₅`.
+-/
+
+open Nat
+
+/-- `a ^ n + 1` is prime only if `n` is a power of two. -/
+theorem pow_of_pow_add_prime (a n : ℕ) (ha : 1 < a) (hn : 1 < n) (hP : Nat.Prime (a ^ n + 1)) :
```

Comment:
```suggestion
theorem pow_of_pow_add_prime {a n : ℕ} (ha : 1 < a) (hn : 1 < n) (hP : Prime (a ^ n + 1)) :
```
Or maybe `(a ^ n + 1).Prime`.

Reply:
Does your suggestion get the right `Prime`?

Reply:
I don't know indeed, but my guess is that is the same proof works then yes (`(...).Prime` should).

Reply:
Done. Thanks for all your reviewing work!